openwisp_ipam/api/utils.py

```diff
@@ -18,3 +22,27 @@ class FilterByOrganizationManaged(FilterByOrganization):
 
     def get_organization_queryset(self, qs):
         return qs.filter(organization__in=self.request.user.organizations_managed)
+
+
+class FilterByParent:
```

Comment:
@nemesisdesign I have created this reusable class that filters the parent's queryset. If the parent does not have an organization accessible by the user (or managed by user, checked via `FilterByParentManaged` below) then this will receive an empty queryset and will then raise an error. Please take a look.

Reply:
Coverage got reduced, here is the new report.

```
Name                                                 Stmts   Miss  Cover   Missing
----------------------------------------------------------------------------------
openwisp_ipam/admin.py                                 132      0   100%
openwisp_ipam/api/responses.py                           4      0   100%
openwisp_ipam/api/serializers.py                        25      0   100%
openwisp_ipam/api/utils.py                              33      3    91%   15, 43, 46
openwisp_ipam/api/views.py                             188      0   100%
openwisp_ipam/apps.py                                    7      0   100%
openwisp_ipam/base/fields.py                            51      1    98%   58
openwisp_ipam/base/forms.py                              3      0   100%
openwisp_ipam/base/models.py                           169      3    98%   225, 230-231
openwisp_ipam/compat.py                                 17     11    35%   11-20, 26, 29-30, 32-33
openwisp_ipam/management/commands/export_subnet.py       4      0   100%
openwisp_ipam/management/commands/import_subnet.py       4      0   100%
openwisp_ipam/models.py                                 10      0   100%
openwisp_ipam/urls.py                                    5      0   100%
----------------------------------------------------------------------------------
TOTAL                                                  652     18    97%
```

The lines not included are all `raise NotImplementedError()`. I don't think we need tests for these here, we can add them later in `openwisp-users`. What do you think?

Reply:
@Purhan it is generally a good idea to test the code where it resides. You can create an instance of `FIlterByParent` class inside a test case and check whether `NotImplementedError` is raised while calling respective methods. 

I would use [subtests](https://www.caktusgroup.com/blog/2017/05/29/subtests-are-best/) to group such tests under one test case.